be/src/Project/Services/Catalog/Catalog.Api/Services/Products/ProductsService.cs

```diff
@@ -285,6 +285,16 @@ public async Task DeleteAsync(DeleteProductServiceModel model)
 
             product.IsActive = false;
 
+            var message = new DeletedProductIntegrationEvent
+            {
+                ProductId = model.Id,
+                Language = model.Language,
+                Username = model.Username,
+                OrganisationId = model.OrganisationId
+            };
+
+            this.eventBus.Publish(message);
```

Comment:
**suggestion:** it should be after await this.catalogContext.SaveChangesAsync(); because if saved to the database fails it makes no sense to publish the event.

Reply:
@dawiddworak88  

You've right